src/actions/filterActions.js

```diff
@@ -14,6 +15,21 @@ export const changeEventFilter = (history) => (filter) => {
   }
 }
 
+export const changeUncorrelatedEventFilter = (history) => (filter) => {
```

Comment:
You can merge this with changeEventFilter by changing the signature of changeEventFilter from (history) => (filter) => {...} to (history, urlLoader = filterService.getUrlFromFilter) => (filter) => {...}

Since the only difference between changeEventFilter and changeUncorrelatedEventFilter is:

filterService.getUrlFromFilter(history, filter)
vs
filterService.getUrlFromUncorrelatedFilter(history, filter)

That line can become

urlLoader (history, filter)

Then the call to changeUncorrelatedEventFilter becomes

dispatch(changeEventFilter(history, filterActions.getUrlFromUncorrelatedFilter)(newFilter))